src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -1149,6 +1165,72 @@ internal bool HasIsByRefLikeAttribute(EntityHandle token)
         }
 
 #nullable enable
+        internal string? GetUnsupportedCompilerFeature(EntityHandle token, IAttributeNamedArgumentDecoder attributeNamedArgumentDecoder, IModuleSymbolInternal containingModule, CompilerFeatureRequiredFeatures allowedFeatures)
+        {
+            string? unsupportedFeature = null;
+            if (!token.IsNil)
+            {
+                unsupportedFeature = GetFirstUnsupportedCompilerFeatureFromToken(token, attributeNamedArgumentDecoder, allowedFeatures);
+            }
+
+            // Check the containing module and assembly as well, if the symbol itself was fine
+            return unsupportedFeature ?? containingModule.GetUnsupportedCompilerFeature() ?? containingModule.ContainingAssembly.GetUnsupportedCompilerFeature();
```

Comment:
>return unsupportedFeature ?? containingModule.GetUnsupportedCompilerFeature() ?? containingModule.ContainingAssembly.GetUnsupportedCompilerFeature(); [](http://example.com/codeflow?start=12&length=149)

As I mentioned earlier, I do not like this bundle.
1. This creates coupling between components.
2. It breaks clear parent/child chain of relationship. And likely leads to some strange behavior. Why do we want to check module/assembly for a parameter and return that value, skipping containing method, and type. 

I think the implementation would more sensible, much clearer and more robust if we moved this part to PE symbols and each symbol would only look at its immediate parent explicitly and only when it is appropriate. For example, there is no need to check parent for parameters and type parameters because implementation is not interested for information about them individually, but only in combination with their parent, which calculates combined result.